cli/command/stack/deploy_composefile.go

```diff
@@ -321,7 +321,16 @@ func deployServices(
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
 			}
-
+			if !updateOpts.QueryRegistry {
+				serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
```

Comment:
I'm thinking these two blocks should look something like this (using a `switch/case` instead of an `if/else if`):

```go
switch {
// image should be updated by the server using QueryRegistry
case resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]):
    updateOpts.QueryRegistry = true
// image hasn't change, so the serviceSpec needs to be updated to use the existing image that was set by QueryRegistry on the previous deploy. Otherwise this will result in an incorrect service update.
case image == service.Spec.Labels[convert.LabelImage]:
    serviceSpec.TaskTemplate.ContainerSpec.Image = service.Spec.TaskTemplate.ContainerSpec.Image
}
```